aopy/preproc/base.py

```diff
@@ -732,3 +732,58 @@ def calc_eye_calibration(cursor_data, cursor_samplerate, eye_data, eye_samplerat
         return coeff, correlation_coeff, cursor_data_aligned, eye_data_aligned
     else:
         return coeff, correlation_coeff
+
+def calc_eye_target_calibration(eye_data, eye_samplerate, event_times, event_codes, target_pos,
+    align_events=range(81,89), trial_end_events=[239], offset=0., duration=0.1, return_datapoints=False):
+    """
+    Extracts eye data and calibrates, aligning eye data and calculating the least square fitting coefficients 
+    between eye positions and target positions
+    
+    Args:
+        eye_data (nt, 2 or 4): eye data in time (optionally for both left and right eyes)
+        eye_samplerate (float): sampling rate of the eye data
+        event_times (nevent): times at which events occur
+        event_codes (nevent): codes for each event
+        target_pos (nevent, 3): target positions (x,y,z) for each event
+        align_events (list, optional): list of event codes to use for alignment. By default, align to
+            when the cursor enters 8 peripheral targets
+        trial_end_events (list, optional): list of end events to use for alignment. By default trial end is code 239
```

Comment:
i recommend using trial_separate instead of get_trial_segments for this alignment, that way you can more easily exclude trials. see the recently updated calc_eye_calibration() function above